src/util/dbid.h

```diff
@@ -0,0 +1,104 @@
+#ifndef MIXXX_DBID_H
+#define MIXXX_DBID_H
+
+#include <ostream>
+
+#include <QString>
+#include <QtDebug>
+
+#include "util/assert.h"
+
+
+// Immutable base class for DB identifiers.
+// Implicit conversion from/to the native value type has been
+// disabled on purpose in favor of explicit conversion!
+class DbId {
+public:
+    typedef int value_type;
```

Comment:
Coincidence ;) I'm already working on this and had the same idea to hide the 
implementing type as far as possible. It will only be exposed by a toInt() 
function for backward compatibility with legacy code.

I will prepare a separate PR that contains just the new DbId and some 
derived classes.

On 08/21/2015 08:22 AM, Daniel Schürmann wrote:
>
> In src/util/dbid.h 
> <https://github.com/mixxxdj/mixxx/pull/592#discussion_r37609025>:
>
> > +#define MIXXX_DBID_H
> > +
> > +#include <ostream>
> > +
> > +#include <QString>
> > +#include <QtDebug>
> > +
> > +#include "util/assert.h"
> > +
> > +
> > +// Immutable base class for DB identifiers.
> > +// Implicit conversion from/to the native value type has been
> > +// disabled on purpose in favor of explicit conversion!
> > +class DbId {
> > +public:
> > +    typedef int value_type;
>
> If yes, the type should be more private.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/592/files#r37609025>.
>